plugins/flashrom/fu-plugin-flashrom.c

```diff
@@ -170,34 +185,44 @@ fu_plugin_update (FuPlugin *plugin,
 		  GError **error)
 {
 	FuPluginData *data = fu_plugin_get_data (plugin);
-	g_autofree gchar *firmware_fn = NULL;
-	g_autofree gchar *tmpdir = NULL;
-	const gchar *argv[] = {
-		data->flashrom_fn,
-		"--programmer", "internal:laptop=force_I_want_a_brick",
-		"--write", "xxx",
-		"--verbose", NULL };
-
-	/* write blob to temp location */
-	tmpdir = g_dir_make_tmp ("fwupd-XXXXXX", error);
-	if (tmpdir == NULL)
-		return FALSE;
-	firmware_fn = g_build_filename (tmpdir, "flashrom-firmware.bin", NULL);
-	if (!fu_common_set_contents_bytes (firmware_fn, blob_fw, error))
+	gsize sz = 0;
+	const guint8 *buf = g_bytes_get_data (blob_fw, &sz);
+
+	if (flashrom_layout_read_from_ifd (&data->layout, data->flashctx, NULL, 0)) {
```

Comment:
That assumes a recent Intel platform. What about other platforms that don't have an IFD?
Can you detect the southbridge vendor and version to conditionally enable IFD support?

Reply:
Sure, we certainly could do.